### ISSUE TITLE
Update `1.29.0`: Migration V1290__auto_main.sql failed - Subquery returns more than 1 row

### ISSUE THREAD

**Issue by sklages** [miso-lims/miso-lims#2516]
Just wanted to update to current version from `1.28.1` and got an error:

```
Migrating schema `miso_seqcore` to version 1290 - auto main
WARNING: DB: Subquery returns more than 1 row (SQL State: 21000 - Error Code: 1242)
ERROR: Migration of schema `miso_seqcore` to version 1290 - auto main failed! Please restore backups and roll back database and code!
ERROR: 
Migration V1290__auto_main.sql failed
-------------------------------------
SQL State  : 21000
Error Code : 1242
Message    : Subquery returns more than 1 row
Location   : db/migration/V1290__auto_main.sql (/project/miso/2018/build_tools/flyway/5.2.4/file:/project/miso/2018/build_tools/flyway/5.2.4/jars/sqlstore-1.29.0.jar!/db/migration/V1290__auto_main.sql)
Line       : 20
Statement  : UPDATE Library SET
index1Id = (
  SELECT link.index_indexId
  FROM Library_Index link
  JOIN Indices ind ON ind.indexId = link.index_indexId
  WHERE link.library_libraryId = Library.libraryId
  AND ind.position = 1
),
index2Id = (
  SELECT link.index_indexId
  FROM Library_Index link
  JOIN Indices ind ON ind.indexId = link.index_indexId
  WHERE link.library_libraryId = Library.libraryId
  AND ind.position = 2
)
```

What's going wrong here?

I am restoring `1.28.1` for now ..

**Comment by djcooke** [comment]
Looks like you have libraries with conflicting indices. Run this on your database to find them:

```
SELECT library_libraryId, ind.position
FROM Library_Index link
JOIN Indices ind ON ind.indexId = link.index_indexId
GROUP BY link.library_libraryId, ind.position
HAVING COUNT(*) > 1
```

Any results mean that the library has multiple indices of the specified position. They'll need fixed before you can upgrade. Here's a query to help you investigate any cases that are returned, as you probably won't be able to view them in MISO:

```
SELECT l.libraryId, l.alias, ind.indexId, ind.name, ind.sequence
FROM Library l
JOIN Library_Index link ON link.library_libraryId = l.libraryId
JOIN Indices ind ON ind.indexId = link.index_indexId
```

You can remove indices from a library using this query. Replace LIBRARY_ID and INDEX_ID appropriately:

```
DELETE FROM Library_Index
WHERE library_libraryId = LIBRARY_ID
AND index_indexId = INDEX_ID;
```

**Comment by sklages** [comment]
Well, thanks for that! 
Indeed, there was one single library from 2014 with two conflicting indices .. it survived so many years :-)
I removed one of these indices just as you proposed.

But I cannot update ti `1.29.0` yet because I noticed the "known issues", specifically `Error when attempting to upload files` .. we are frequently uploading Excel files and HTML reports .. I will wait for `1.30.0` to be released and report back.
